triples-generator/src/main/scala/io/renku/triplesgenerator/events/categories/triplesgenerated/projectinfo/CommitAuthorFinder.scala

```diff
@@ -21,61 +21,46 @@ package io.renku.triplesgenerator.events.categories.triplesgenerated.projectinfo
-import io.renku.http.client.UrlEncoder.urlEncode
-import io.renku.http.client.{AccessToken, RestClient}
+import io.renku.graph.model.{persons, projects}
+import io.renku.http.client.{AccessToken, GitLabClient}
 import io.renku.triplesgenerator.events.categories.ProcessingRecoverableError
 import io.renku.triplesgenerator.events.categories.triplesgenerated.RecoverableErrorsRecovery
+import org.http4s.implicits.http4sLiteralsSyntax
 import org.http4s.{EntityDecoder, InvalidMessageBodyFailure, Request, Response, Status}
 import org.typelevel.log4cats.Logger
 
-import scala.concurrent.duration.{Duration, FiniteDuration}
-
 private trait CommitAuthorFinder[F[_]] {
   def findCommitAuthor(projectPath: projects.Path, commitId: CommitId)(implicit
       maybeAccessToken:             Option[AccessToken]
   ): EitherT[F, ProcessingRecoverableError, Option[(persons.Name, persons.Email)]]
 }
 
 private object CommitAuthorFinder {
-  def apply[F[_]: Async: Logger](gitLabThrottler: Throttler[F, GitLab]): F[CommitAuthorFinder[F]] = for {
-    gitLabUrl <- GitLabUrlLoader[F]()
-  } yield new CommitAuthorFinderImpl(gitLabUrl.apiV4, gitLabThrottler)
+  def apply[F[_]: Async: Logger](gitLabClient: GitLabClient[F]): F[CommitAuthorFinder[F]] = new CommitAuthorFinderImpl(
+    gitLabClient
+  ).pure[F].widen
 }
 
 private class CommitAuthorFinderImpl[F[_]: Async: Logger](
-    gitLabApiUrl:           GitLabApiUrl,
-    gitLabThrottler:        Throttler[F, GitLab],
-    recoveryStrategy:       RecoverableErrorsRecovery = RecoverableErrorsRecovery,
-    retryInterval:          FiniteDuration = RestClient.SleepAfterConnectionIssue,
-    maxRetries:             Int Refined NonNegative = RestClient.MaxRetriesAfterConnectionTimeout,
-    requestTimeoutOverride: Option[Duration] = None
-) extends RestClient(gitLabThrottler,
-                     retryInterval = retryInterval,
-                     maxRetries = maxRetries,
-                     requestTimeoutOverride = requestTimeoutOverride
-    )
-    with CommitAuthorFinder[F] {
+    gitLabClient:     GitLabClient[F],
+    recoveryStrategy: RecoverableErrorsRecovery = RecoverableErrorsRecovery
+) extends CommitAuthorFinder[F] {
 
   import org.http4s.Method.GET
   import org.http4s.dsl.io.{NotFound, Ok}
 
   override def findCommitAuthor(path: projects.Path, commitId: CommitId)(implicit
       maybeAccessToken:               Option[AccessToken]
   ): EitherT[F, ProcessingRecoverableError, Option[(persons.Name, persons.Email)]] = EitherT {
-    {
-      for {
-        projectsUri <- validateUri(s"$gitLabApiUrl/projects/${urlEncode(path.value)}/repository/commits/$commitId")
-        maybeAuthor <- send(secureRequest(GET, projectsUri))(mapTo[(persons.Name, persons.Email)])
-      } yield maybeAuthor
-    }.map(_.asRight[ProcessingRecoverableError]).recoverWith(recoveryStrategy.maybeRecoverableError)
+    gitLabClient
+      .send(GET, uri"projects" / path.value / "repository" / "commits" / commitId.show, "commit")(
```

Comment:
I think we used the `commit-details` name for that API in the other services. Maybe it would be good to keep the same names for the same APIs.